lib/addons/kubernetes-ingress/index.ts

```diff
@@ -0,0 +1,213 @@
+     * @default "3600"
+     */
+    idleTimeout?: string;
+}
+
+// Set default properties for the add-on
+const defaultProps: KubernetesIngressAddOnProps = {
+    name: "kubernetes-ingress",
+    chart: "ingress-nginx",
+    release: "k8s-ingress",
+    version: "4.10.0",
+    repository: "https://kubernetes.github.io/ingress-nginx",
+    backendProtocol: 'http',
+    crossZoneEnabled: true,
+    internetFacing: true,
+    targetType: 'ip',
+    namespace: 'kube-system',
+    idleTimeout: '3600'
+};
+
+// Define the class for the Kubernetes Ingress Add-On, extending HelmAddOn
+@supportsALL
+export class KubernetesIngressAddOn extends HelmAddOn {
+    private readonly options: KubernetesIngressAddOnProps;
+
+    // Constructor for the class, merging default props with user-defined props
+    constructor(props?: KubernetesIngressAddOnProps) {
+        super({ ...defaultProps, ...props } as HelmAddOnProps);
+        this.options = { ...defaultProps, ...props } as KubernetesIngressAddOnProps;
+    }
+
+    // Dependency decorator to ensure this add-on is deployed after the AWS Load Balancer Controller
+    @dependable(AwsLoadBalancerControllerAddOn.name)
+    async deploy(clusterInfo: ClusterInfo): Promise<Construct> {
+        const props = this.options;
+
+
+
+        // Setup service annotations based on the properties provided
+        const presetAnnotations: any = {
+            'service.beta.kubernetes.io/aws-load-balancer-backend-protocol': props.backendProtocol,
+            'service.beta.kubernetes.io/aws-load-balancer-attributes': `load_balancing.cross_zone.enabled=${props.crossZoneEnabled}`,
+            'service.beta.kubernetes.io/aws-load-balancer-scheme': props.internetFacing ? 'internet-facing' : 'internal',
+            'service.beta.kubernetes.io/aws-load-balancer-type': 'external',
+            'service.beta.kubernetes.io/aws-load-balancer-nlb-target-type': props.targetType,
+            'external-dns.alpha.kubernetes.io/hostname': props.externalDnsHostname,
+            'service.beta.kubernetes.io/aws-load-balancer-connection-idle-timeout': '3600'
+            
+            
+        };
+
+        // Define values for Helm chart based on properties and annotations
+        const values: Values = {
+            controller: {
+                service: {
+                    annotations: presetAnnotations
+                },
+                ingressClassResource: {
+                    name: props.ingressClassName || "nginx",
+                    enabled: true,
+                    default: props.isDefaultClass ?? false,
+                    controllerValue: props.controllerClass || "k8s.io/ingress-nginx"
+                },
+                electionID: props.electionId || "ingress-controller-leader"
+            }
+        };
+
+        // Configure SSL-related annotations if certificate resource name is provided
+        if (props.certificateResourceName) {
```

Comment:
Style: the code block on line 188-196 is almost identical to the code block in lines 196-200. 

How about you use this logic:

```ts
let certificateResourceARN = props.certificateResourceARN;

if(!certificateResourceARN && props.certificateResourceName) {
   certificateResourceARN = <populate from the resource provider>;
}

if(certificateResourceARN) {
   // set annotations
}
```